salt/fileserver/__init__.py

```diff
@@ -492,17 +492,17 @@ def clear_lock(self, back=None, remote=None):
                 errors.extend(bad)
         return cleared, errors
 
-    def update(self, back=None):
+    def update(self, back=None, **kwargs):
```

Comment:
Please update this signature to include the specific arguments that should be passed

Reply:
The arguments are backend specific. How to handle this?

Another solution would be to pass this `remotes` arg to all backend, but it doesn't always makes sense (for example the `roots` backend).

Currently logic throws an error when the parameter is incorrect. Isn't this sufficient?

Reply:
I've added a test for this.

Reply:
Can we be more specific on which kwargs to accept?
If not, why not.

Reply:
Guess I'm late to the discussion...

Reply:
Not merged yet :trollface: